xbmc/platform/linux/input/LinuxInputDevices.cpp

```diff
@@ -1045,7 +1045,18 @@ void CLinuxInputDevice::GetInfo(int fd)
   else if (m_deviceType & LI_DEVICE_MOUSE)
     m_devicePreferredId = LI_DEVICE_MOUSE;
   else
-    m_devicePreferredId = LI_DEVICE_NONE;
+  {
+    /* if no better fit and have at least 1 key */
+    if (num_keys >= 1)
+    {
+      m_deviceType |= LI_DEVICE_KEYBOARD;
+      m_deviceCaps |= LI_CAPS_KEYS;
+
+      m_deviceMinKeyCode = 0;
+      m_deviceMaxKeyCode = 127;
+    }
+    else
+      m_devicePreferredId = LI_DEVICE_NONE;
 
```

Comment:
Missing } here. Stylistically I'd remove the initial { and have 
```else if (num_keys >= 1)```

Reply:
you're correct, I've made the change as requested

Reply:
We are not setting m_devicePreferredId in this code path - is that the desired behaviour?
But we are setting LI_DEVICE_KEYBOARD bit of m_deviceType which means if this function is called again
we will assign ``m_devicePreferredId = LI_DEVICE_KEYBOARD`` which seems questionable.
But I'm not really familiar with this code - @garbear can probably comment more accurately.

Reply:
sorry you are correct that was not the desired behavior, from my understand we need to both bitwise or the value into device type and set the m_devicePreferredId, i have ammended once again.

Reply:
I'm still unsure about this. I assume this function gets called for each input device.
e.g. keyboard/remote/joystick/mouse
If you had a touchscreen and a single shutdown key attached do you want the preferred device to be set to keyboard?

Currently the logic means that if any device is a real (20+ keys) keyboard it will be preferred.
Followed by touchscreen, remote, joystick then mouse.

I wonder if the "small keyboard" should have a unique ``LI_DEVICE_KEYS`` category that fall below the other input devices in precedence.